### PR TITLE
CIS2 Tokens List & Details Sidebar

### PR DESCRIPTION
**Pull request by parv0888**
## Purpose

Added CIS2 Tokens List Page & Details Sidebar

## Changes

Added CIS2 Tokens List Page & Details Sidebar

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

## CLA acceptance

By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [x] I accept the above linked CLA.

### CODE REVIEW

**Comment by parv0888** [comment]
image

**Comment by parv0888** [comment]
Pending : 
- Showing Token Metadata & Accounts in Token Details Sidebar
- Showing Token Create Time and Last Updated Time
- Showing Token Transactions summary in Token Details Sidebar

File: frontend/src/components/Navigation/Navigation.vue [hunk 1]
```diff
@@ -36,5 +36,9 @@ const routes: Route[] = [
 		title: 'Nodes',
 		path: '/nodes',
 	},
+	{
+		title: 'Tokens',
```
Comment (abizjak) [inline frontend/src/components/Navigation/Navigation.vue:40]: Shouldn't this indicate it's CIS2 tokens?

File: frontend/src/pages/tokens/index.vue [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+			<TableBody>
+				<TableRow
+					v-for="token in tokens"
+					:key="token.contractIndex + token.tokenId"
+				>
+					<TableTd>
+						<TokenLink :data="token.tokenId" :url="token.metadataUrl" />
+					</TableTd>
+					<TableTd>
+						{{ token.contractIndex }} / {{ token.contractSubIndex }}
```
Comment (abizjak) [inline frontend/src/pages/tokens/index.vue:25]: We should display the "TokenAddress" somewhere https://github.com/Concordium/concordium-base/blob/70c1c88250dc65a56bca46965c2df9a278a5703e/rust-src/concordium_base/src/cis2_types.rs#L91

This contains index + subindex + tokenId in a single copyable string. See https://github.com/Concordium/concordium-base/blob/70c1c88250dc65a56bca46965c2df9a278a5703e/rust-src/concordium_base/src/cis2_types.rs#L1195 for examples.

File: frontend/src/pages/tokens/index.vue [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+							:length="25"
+							:suffix="'...'"
+						/>
+					</TableTd>
+
+					<TableTd v-if="breakpoint >= Breakpoint.SM" align="right">
+						<span class="numerical">
+							<TokenAmount
+								:symbol="token.metadata?.symbol"
+								:amount="token.totalSupply"
```
Comment (abizjak) [inline frontend/src/pages/tokens/index.vue:41]: Maybe I missed it , but you don't seem to be using the "unique" propery to show if this is fungible or not anywhere.

Comment (parv0888) [reply frontend/src/pages/tokens/index.vue:41]: its not being displayed in the token details sidebar

File: frontend/src/components/Navigation/Navigation.vue (hunk 1 above)
Comment (parv0888) [reply frontend/src/components/Navigation/Navigation.vue:40]: done

File: frontend/src/pages/tokens/index.vue (hunk 2 above)
Comment (parv0888) [reply frontend/src/pages/tokens/index.vue:29]: We attempted this. However there seems to be now way of knowing the type of token id. All the responses we get back from the node return token id as `string`. The actual type of token if `u128`, `u64` etc. Cannot be inferred from the token events. 

Can you help us in finding a way to decipher the type of Token Id?

**Comment by parv0888** [comment]
image

File: frontend/src/pages/tokens/index.vue (hunk 2 above)
Comment (abizjak) [reply frontend/src/pages/tokens/index.vue:29]: I don't understand. For the token address it does not matter whether token id is a u128 or u16 or () internally by the contract.

The token id format is specified by the standard and it serialization is fixed. What you get back is probably a hex string with the serialization, and that is all you need to construct the token address.

**Pull request closed without merging**